### PR TITLE
CVE-2014-0116

### PR DESCRIPTION
**Pull request by gavrielrh**
Initial draft of CVE-2014-0116

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,30 @@
+GEM
```
Comment (MattThyng) [inline Gemfile.lock]: You should remove the Gemfile.lock from git

File: cves/CVE-2014-0116.yml [hunk 2]
```diff
@@ -205,4 +223,8 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those in the software
     engineering industry would find interesting.
-  answer:
+  answer: |
```
Comment (MattThyng) [inline cves/CVE-2014-0116.yml:225]: How can vulnerabilities like this one be prevented in the future?

File: cves/CVE-2014-0116.yml [hunk 3]
```diff
@@ -45,7 +45,11 @@ description_instructions: |
   that outsiders to Struts would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  A fix introduced in a previous patch (2.3.16.2) to secure the getClass() method
+  didn't account for the wildcard operator (*) being used when configuring the
+  cookiesName param. This opened up the possibility for attackers to manipulate
+  the class loader and modify session state.
```
Comment (MattThyng) [inline cves/CVE-2014-0116.yml]: I would be a little more general with this description.

File: Gemfile.lock (hunk 1 above)
Comment (gavrielrh) [reply Gemfile.lock]: 👍 Thank you; Not sure how that snuck by me on this PR. I've removed it as of the latest commit.

File: cves/CVE-2014-0116.yml (hunk 2 above)
Comment (gavrielrh) [reply cves/CVE-2014-0116.yml:225]: I've added additional info about prevention to my PR itself in the latest commit.
In short, vulnerabilities like this one can be prevented by doing due diligence regarding researching pre-existing sanitization solutions (in this case, within the same repo), by communicating with team-members about such issues, and by creating more robust test cases.

File: cves/CVE-2014-0116.yml (hunk 3 above)
Comment (gavrielrh) [reply cves/CVE-2014-0116.yml]: Thank you for the feedback; I've generalized my description so it doesn't specify the patch, nor the specific methods used. I also simplified the explanation for how the vulnerability might be exploited.

File: cves/CVE-2014-0116.yml [hunk 4]
```diff
@@ -45,7 +45,10 @@ description_instructions: |
   that outsiders to Struts would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  A fix applied in a previous update didn't account for the wildcard (*) operator.
+  By missing this operator in the appropriate blacklist for disallowed inputs, this 
+  allowed attackers to bypass the safeguards in place and craft exploits for the system.
```
Comment (kar5254) [inline cves/CVE-2014-0116.yml:51]: It may be a good idea to give one or two examples of exploits for this vulnerability. Great description otherwise!

**Pull request merged**